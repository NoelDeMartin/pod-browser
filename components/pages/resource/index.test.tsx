import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { useRouter } from "next/router";

import { useRedirectIfLoggedOut } from "../../../src/effects/auth";
import IndexPage from "./index";

jest.mock("../../../src/effects/auth");
jest.mock("next/router");

describe("Index page", () => {
  beforeEach(() => {
    (useRouter as jest.Mock).mockImplementation(() => {
      return {
        query: {
          iri: encodeURIComponent("https://mypod.myhost.com"),
        },
      };
    });
  });

  test("Renders a logout button", () => {
    const tree = shallow(<IndexPage />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  test("Redirects if the user is logged out", () => {
    shallow(<IndexPage />);
    expect(useRedirectIfLoggedOut).toHaveBeenCalled();
  });
});
