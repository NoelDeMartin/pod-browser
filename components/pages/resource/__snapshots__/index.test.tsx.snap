// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page Renders a logout button 1`] = `
<WithStyles(ForwardRef(Container))>
  <Container
    iri="https://mypod.myhost.com"
  />
</WithStyles(ForwardRef(Container))>
`;
